test-framework/junit5-internal/src/main/java/io/quarkus/test/QuarkusUnitTest.java

```diff
@@ -125,6 +125,9 @@ private QuarkusUnitTest(boolean useSecureConnection) {
     }
 
     public QuarkusUnitTest assertException(Consumer<Throwable> assertException) {
+        if (this.assertException != null) {
+            throw new IllegalStateException("Don't set the asserted or excepted exception twice.");
```

Comment:
```suggestion
            throw new IllegalStateException("Don't set the asserted or excepted exception twice"
                    + " to avoid shadowing out the first call.");
```